File: 03/sum_fibonacci.hs
-- n番目までのフィボナッチ数の和を求める
-- 関数sumfib

fib :: Int -> Int
fib 0 = 0 -- 0番目のフィボナッチ数
fib 1 = 1 -- 1番目のフィボナッチ数
fib n = fib (n-1) + fib (n-2) 
  -- n番目のフィボナッチ数（ただしn>=2）

sumfib :: Int -> Int
sumfib n = sum [fib x | x <- [0..n]]

main = print (sumfib 10)

File: 03/sum_fibonacci.py
# n番目までのフィボナッチ数の和を求める
# 関数sumbfib

def fib(n: int) -> int:
    if n == 0:
        return 0
    elif n == 1:
        return 1
    else:
        return fib(n-1) + fib(n-2)

def sum_fib(n: int) -> int:
    return sum([fib(i) for i in range(n+1)])

print(sum_fib(10))
